internal/provider/data_source_http.go

```diff
@@ -162,12 +163,29 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 
 	caCertificate := model.CaCertificate
 
-	tr := &http.Transport{
-		TLSClientConfig: &tls.Config{},
+	tr, ok := http.DefaultTransport.(*http.Transport)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Error configuring http transport",
+			"Error http: Can't configure http transport.",
+		)
+		return
+	}
+
+	// Prevent issues with tests modifying shared transport.
```

Comment:
Nit: I think these issues could also happen with multiple, differing data source TLS configurations as well 👍 

```suggestion
	// Prevent issues with multiple data source configurations modifying the shared transport.
```